python/tank/authentication/saml2_sso.py

```diff
@@ -0,0 +1,605 @@
+# Copyright (c) 2017 Autodesk.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+"""
+Module to support SSO login via a web browser and automated session renewal.
+
+This module, in conjunction with the slmodule, handles authentication on
+SSO enabled Shotgun websites. When we are in a session connected via SSO, we
+also handle automatic (and headless) renewal of the session.
+"""
+
+import base64
+import json
+import logging
+import os
+import time
+
+from .session import Session
+from .ui.qt_abstraction import QtCore, QtNetwork, QtGui, QtWebKit
```

Comment:
do we have full coverage for all these modules in all our DCCs etc? @jfboismenu should we be worried about `QtWebKit` for example? I know a lot of QT distros don't ship will all qt modules included.